data-pipeline/content-cache-updater/src/main/java/org/ekstep/ep/samza/service/ContentCacheUpdaterService.java

```diff
@@ -0,0 +1,153 @@
+        this.contentStoreConnection = redisConnect.getConnection(contentStoreDb);
+        this.dialCodeStoreConnection = redisConnect.getConnection(dialCodeStoreDb);
+        this.contentModelListTypeFields = config.getList("contentModel.fields.listType", new ArrayList<>());
+    }
+
+    public void process(ContentCacheUpdaterSource source, ContentCacheUpdaterSink sink) {
+        Map<String, Object> message = source.getMap();
+        String nodeUniqueId = (String) message.get("nodeUniqueId");
+        String objectType = (String) message.get("objectType");
+        Map<String, Object> parsedData = null;
+        if (nodeUniqueId == null || objectType == null) return;
+        LOGGER.info("", "processing event for nodeUniqueId: " + nodeUniqueId);
+        try {
+            if (!nodeUniqueId.isEmpty() && objectType.equalsIgnoreCase("DialCode")) {
+                parsedData = updateDialCodeCache(message, sink);
+                redisConnection=dialCodeStoreConnection;
+                storeId=dialCodeStoreDb;
+            } else if (!nodeUniqueId.isEmpty()) {
+                parsedData = updateContentCache(message, sink);
+                redisConnection=contentStoreConnection;
+                storeId=contentStoreDb;
+            }
+            if(null != parsedData) {
+                redisConnect.addJsonToCache(nodeUniqueId, gson.toJson(parsedData), redisConnection);
+                sink.success();
+            }
+        }catch (JedisException ex) {
+            metrics.incCacheErrorCounter();
+            LOGGER.error("", "Exception when adding to dialcode redis cache", ex);
+            redisConnect.resetConnection();
+            try (Jedis redisConn = redisConnect.getConnection(storeId)) {
+                this.redisConnection = redisConn;
+                if (null != parsedData)
+                    redisConnect.addJsonToCache(nodeUniqueId, gson.toJson(parsedData), redisConnection);
+            }
+        }
+
+    }
+
+    public Map<String, Object> updateDialCodeCache(Map<String, Object> message, ContentCacheUpdaterSink sink) {
+        String nodeUniqueId = (String) message.get("nodeUniqueId");
+        Map<String, Object> parsedData = null;
+            String contentNode = dialCodeStoreConnection.get(nodeUniqueId);
+            if (contentNode != null) {
+                parsedData = gson.fromJson(contentNode, mapType);
+            } else {
+                parsedData = new HashMap<>();
+            }
+            Map<String, Object> newProperties = extractProperties(message);
+            parsedData.putAll(newProperties);
+            if (parsedData.size() > 0) {
+                parsedData.values().removeAll(Collections.singleton(null));
+                parsedData.values().removeAll(Collections.singleton(""));
+            }
+        return parsedData;
+    }
+
+    @SuppressWarnings("unchecked")
+    public Map<String, Object> updateContentCache(Map<String, Object> message, ContentCacheUpdaterSink sink) {
```

Comment:
This should be named as getContentCacheData() as the update is not happening in this method. Most of the code is common along with updateDialCodeCache and can be merged into one method